### PR TITLE
script to take down matrix service api, lambdas, and job queue

### PR DESCRIPTION
**Pull request by parthshahva**
Big red button script for https://github.com/HumanCellAtlas/matrix-service/issues/125

### CODE REVIEW

File: scripts/take_down_deployment_be_careful.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        cdn = cdn_non_prod_template.replace("$DEPLOYMENT_STAGE", args.environment)
+    api_gateway_client.delete_domain_name(
+        domainName=cdn
+    )
+    print(f"took down cdn {cdn}")
+
+    # Throttle all involved lambdas
+    for lambda_template in lambda_name_templates:
+        lambda_name = lambda_template.replace("$DEPLOYMENT_STAGE", args.environment)
+        lambda_client.put_function_concurrency(FunctionName=lambda_name, ReservedConcurrentExecutions=0)
```
Comment (mckinsel) [inline scripts/take_down_deployment_be_careful.py]: The AWS docs say "Note that Lambda automatically reserves a buffer of 100 concurrent executions for functions without any reserved concurrency limit.". Does setting this to zero just mean that the functions can use the account-wide limit instead?

Comment (mckinsel) [reply scripts/take_down_deployment_be_careful.py]: :+1:

Comment (parthshahva) [reply scripts/take_down_deployment_be_careful.py]: After invoking this function with value of 0, if you go the UI, you will see the message "Function is throttled" at the top

File: scripts/take_down_deployment_be_careful.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+
+def main(args):
+    # Take down API Gateway
+    if args.environment == "prod":
+        cdn = cdn_prod
+    else:
+        cdn = cdn_non_prod_template.replace("$DEPLOYMENT_STAGE", args.environment)
+    api_gateway_client.delete_domain_name(
+        domainName=cdn
```
Comment (mckinsel) [inline scripts/take_down_deployment_be_careful.py]: Is it possible for someone to continue to access the API without the custom domain name?

**Comment by mckinsel** [comment]
Looks good. One question: a big part of the current API is just downloading S3 URLs. Those all stay intact after this script is run. Do we want to add a little section that revokes `s3:GetObject` on the results bucket?

File: scripts/take_down_deployment_be_careful.py (hunk 2 above)
Comment (parthshahva) [reply scripts/take_down_deployment_be_careful.py]: I don't think this is possible. Custom domain name apigateway record is the link between an api gateway stage and route 53. As far as I know, you can't access an api gateway stage without this routing setup

Comment (parthshahva) [reply scripts/take_down_deployment_be_careful.py]: ^I may be wrong here. I didn't know about api gateway invoke urls... Looking into this

Comment (calvinnhieu) [reply scripts/take_down_deployment_be_careful.py]: As @mckinsel brought up, the API is still available via the generated invoke URL `<stage_id>.execute-api.us-east-1.amazonaws.com`  We could attach a [Resource Policy](https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-control-access-policy-language-overview.html) to the API denying `invoke` if we want to shut this down as well.

**Pull request closed without merging**